components/IndoorDirections/TypesOfDirections/DifferentBuildingDirections.js

```diff
@@ -0,0 +1,143 @@
+        }
+    });
+
+    const linesToExit = [];
+    pathToClass.forEach((element, index) => {
+        if (index < pathToClass.length - 1) {
+            linesToExit.push({x1: rooms[element].nearestPoint.x, y1: rooms[element].nearestPoint.y, x2: rooms[getNextRoomClass(index)].nearestPoint.x, y2: rooms[getNextRoomClass(index)].nearestPoint.y})
+        }
+    });
+    
+
+    // Go to elevator
+    if(props.floor == getFloorNumber(props.from)) {
+        return(
+            <G>
+                <Line x1={rooms[props.from].x} y1={rooms[props.from].y} x2={rooms[props.from].nearestPoint.x} y2={rooms[props.from].nearestPoint.y} stroke="blue" strokeWidth="5"/>
+                {
+                    linesToElevator.map(el => <Line x1={el.x1} y1={el.y1} x2={el.x2} y2={el.y2} stroke="blue" strokeWidth="5"/>)
+                }
+                <Line x1={rooms["elevator"].x} y1={rooms["elevator"].y} x2={rooms["elevator"].nearestPoint.x} y2={rooms["elevator"].nearestPoint.y} stroke="blue" strokeWidth="5"/>
+            </G>
+        )
+    } 
+    if(props.floor == 1) {
+        return(
+            <G>
+                <Line x1={rooms["elevator"].x} y1={rooms["elevator"].y} x2={rooms["elevator"].nearestPoint.x} y2={rooms["elevator"].nearestPoint.y} stroke="blue" strokeWidth="5"/>
+                {
+                    linesToExit.map(el => <Line x1={el.x1} y1={el.y1} x2={el.x2} y2={el.y2} stroke="blue" strokeWidth="5"/>)
+                }
+            </G>
+        )
+    }
+    return(
+        <G></G>
+    )
+}
+
+/**
+ * Get the floor number of a specific classroom. 
+ * Ex: H829 returns 8
+ * @param {*} name | classroom name
+ */
+export const getFloorNumber = (name) => {
+    if (name) {
+        const num = name.match(/\d+/g).map(Number);
+
+        if(num.length > 1) {
+            return num[0].toString();
+        }
+        if (num.toString().length == 3) {
+            return num.toString().charAt(0)
+        }
+        if (num.toString().length == 4) {
+            return num.toString().charAt(0) + num.toString().charAt(1);
+        }   
+    }
+}
+
+const dijkstra = (graph, start, end) => {
+    // track the lowest cost to reach each node
+    let costs = {};
+    costs[end] = "Infinity";
+    costs = Object.assign(costs, graph[start]);
+
+    // track paths
+    const parents = {end: null};
+    for (let child in graph[start]) {
+        parents[child] = start;
+    }
+
+    const processed = [];
+    let node = lowestCostNode(costs, processed);
+
+    while (node) {
+        let cost = costs[node];
+        let children = graph[node];
+        for (let n in children) {
+            if (String(n) !== String(start)) {
+                let newCost = cost + children[n];
+                if (!costs[n] || costs[n] > newCost) {
+                    costs[n] = newCost;
+                    parents[n] = node;
+                }
+            }
+        }
```

Comment:
Very interesting how the dijkstra algorithm plays out, could you briefly comment in the code the functionality of the algo so its easier to for other dev, to understand? What does the *cost* represents here?

Reply:
Yes, I explain the algorithm inside this [sprike](https://github.com/mohanadarafe/GuideMe/issues/108) in the comments.